satellite.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/urfave/cli/v2"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+)
+
+// SatellitePingResponse is used for un-marshaling JSON document
+// returned from Satellite server
+type SatellitePingResponse struct {
+	Results struct {
+		Katello struct {
+			Services struct {
+				Candlepin struct {
+					Status     string `json:"status"`
+					DurationMs string `json:"duration_ms"`
+				} `json:"candlepin"`
+			} `json:"services"`
+			Status string `json:"status"`
+		} `json:"katello"`
+	} `json:"results"`
+}
+
+// ConfigureSatelliteResult is structure holding information about results
+// of configuring host from Satellite server. The result could be printed
+// in machine-readable format.
+type ConfigureSatelliteResult struct {
+	Hostname                 string `json:"hostname"`
+	HostnameError            string `json:"hostname_error,omitempty"`
+	IsServerSatellite        bool   `json:"is_server_satellite_running"`
+	SatelliteServerHostname  string `json:"satellite_server_hostname"`
+	SatelliteServerScriptUrl string `json:"satellite_server_script_url"`
+	HostConfigured           bool   `json:"host_configured"`
+	format                   string
+}
+
+// Error implement error interface for structure ConfigureSatelliteResult
+func (result ConfigureSatelliteResult) Error() string {
+	var msg string
+	switch result.format {
+	case "json":
+		data, err := json.MarshalIndent(result, "", "    ")
+		if err != nil {
+			return err.Error()
+		}
+		msg = string(data)
+	case "":
+		break
+	default:
+		msg = "error: unsupported document format: " + result.format
+	}
+	return msg
+}
+
+// SatelliteHTTPClient represents http httpClient used for communication with
+// satellite server
+type SatelliteHTTPClient struct {
+	httpClient   *http.Client
+	satelliteURL *url.URL
+}
+
+// NewSatelliteClient creates instance of SatelliteHTTPClient and
+// configure it to use HTTPS
+func NewSatelliteClient(satelliteURL *url.URL) *SatelliteHTTPClient {
+	satClient := SatelliteHTTPClient{}
+	// We have to use insecure HTTPs connection, because most of the customers use
+	// self-signed certificates
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: true,
+	}
```

Comment:
So we're downloading a shell script from a URL and executing it as root, *and* we're downloading that shell script from an unsigned HTTPS endpoint. 😱 

Do customers actually use self-signed certificates? Or do they use certificates that are signed by an internal/private signing authority? Could we expose the option to add a CA root certificate as part of this TLS configuration so that customers can at least use their own signing authority? And maybe make them opt-in to fully disabling the certificate verification?